### PR TITLE
upgrade_shapely_2.0_fix

### PR DESCRIPTION
**Pull request by emanuel-schmid**
Changes proposed in this PR:
- use the `.geoms` attribute as `mask` argument in case of `MultiPolygon`

This PR fixes https://github.com/CLIMADA-project/climada_petals/issues/79

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [x] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by emanuel-schmid** [comment]
@tovogt Excellent! 🥇 Many thanks for the profound analysis. Your suggestions make a lot of sense. I think I'm just going to implement them.

**Comment by emanuel-schmid** [comment]
I've reversed the check for `MultiPolygon` and made it officially the responsibility of the caller to make sure the `geometry` argument is a `list of shapely.geometry`. That is, I updated the doc strings, but not the methods.
This is now the case for 5 related methods.

While at it, I changed the default values for many arguments in these methods to `None` where `False` seemed less appropriate.

I didn't change the argument names, reluctant to alter the signature without being forced to. 🤷 

Bottom line: nothing has changed, the PR is reduced to cosmetics.

**Review by tovogt — COMMENTED** [review]
I would have a tiny suggestion how to clean up the code. But since it only affects the tests, it's not really important. The rest is fine with me.

File: climada/entity/exposures/test/test_nightlight.py [hunk 1]
```diff
@@ -125,25 +125,27 @@ def test_check_nl_local_file_exists(self):
...
-        self.assertIn('No satellite files found locally in climada/entity/exposures', cm.output[0])
+        self.assertIn(f'No satellite files found locally in {Path("climada/entity/exposures")}',
+                      cm.output[0])
       
         # test raises with wrong path
         with self.assertRaises(ValueError) as cm:
             nightlight.check_nl_local_file_exists(check_path = '/random/wrong/path')
-        self.assertEqual('The given path does not exist: /random/wrong/path', str(cm.exception))
+        self.assertEqual(f'The given path does not exist: {Path("/random/wrong/path")}',
+                         str(cm.exception))
```
[the remark this answers is not in the document]
Comment (emanuel-schmid) [reply climada/entity/exposures/test/test_nightlight.py:148]: Right, that's better.

**Comment by tovogt** [comment]
Thanks, looks good!
